WordPress/src/main/java/org/wordpress/android/ui/posts/PostsListActivity.java

```diff
@@ -48,11 +55,17 @@ public void onCreate(Bundle savedInstanceState) {
         } else {
             mSite = (SiteModel) savedInstanceState.getSerializable(WordPress.SITE);
         }
+
+        if (mSite == null) {
+            mSite = mSiteStore.getSiteByLocalId(AppPrefs.getSelectedSite());
```

Comment:
Ah ok, that's a valid point. But I think that makes sense to show the PostList of the site where that post was published (ie pass the model in the intent).

Also, I think we opened the PostList and not the post editor for only one reason: the legacy editor opened published post in HTML mode only, that made the experience pretty poor. But now we have the new editor, we should open the post editor instead of the post list in that intent (that would solve my original concern at the same time).